akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala

```diff
@@ -152,12 +151,16 @@ class Http2ServerDemux extends GraphStage[BidiShape[Http2SubStream, FrameEvent,
 
               dispatchSubstream(Http2SubStream(frame, data))
 
+            case PriorityFrame(streamId, exclusiveFlag, streamDependency, weight) ⇒
+              // must be before the "unknown stream id" case, since can be sent eagerly (e.g. firefox does so)
+              debug(s"Received PriorityFrame for stream $streamId with ${if (exclusiveFlag) "exclusive " else "non-exclusive "} dependency on stream $streamDependency and weight $weight")
+
             case e: StreamFrameEvent if !incomingStreams.contains(e.streamId) ⇒
               // if a stream is invalid we will GO_AWAY
-              pushGOAWAY()
+              pushGOAWAY(ErrorCode.PROTOCOL_ERROR, "Unknown stream id: " + e.streamId)
 
             case h: ParsedHeadersFrame ⇒
-              pushGOAWAY()
+              pushGOAWAY(ErrorCode.PROTOCOL_ERROR, "Unexpected internal frame reached Demux! Was: " + h)
```

Comment:
The error message here is not quite correct. When we receive it here it is probably a trailing header frame here which we currently don't support but which we should ignore. It's ok to leave it like this but add a TODO comment to fix it in the course of #531.